bug(test): file round-trip-mismatch test is brittle to $TMPDIR (hardcodes /tmp)

## Problem

`tests/test_cmd_file.py::test_file_inline_body_staging_round_trip_mismatch_errors` fails when `$TMPDIR` is set to anything other than exactly `/tmp` (e.g. `/tmp/claude-1000` inside the Claude Code sandbox). It passes in environments where `TMPDIR` is unset and `tempfile.gettempdir()` resolves to `/tmp`.

```
assert rc != 0  →  AssertionError: 0 != 0
tests/test_cmd_file.py:330
```

This is a **brittle test, not a production defect.** The inline-`--body` round-trip fence in the `file` command (`skills/jared/scripts/jared:610-630`) is intact and correct — it stages the body to `tempfile.NamedTemporaryFile(...)`, reads it back, and refuses (rc 2, no `gh issue create`) on mismatch. No corrupt issue can land.

## Root cause

The test simulates silent staging corruption by monkeypatching `Path.read_text` to return `""` — but it scopes the interception to the staged temp file via:

```python
if self.parent == Path("/tmp") and self.suffix == ".md":
    return ""
```

The staged file is created by `tempfile.NamedTemporaryFile`, which honors `tempfile.gettempdir()` / `$TMPDIR`. When `$TMPDIR=/tmp/claude-1000`, the staged file's parent is `/tmp/claude-1000`, so `self.parent == Path("/tmp")` is `False`, the truncating read-back never fires, the fence sees a matching body, and the command returns rc 0 — failing the test's `assert rc != 0`.

So the test only exercises the fence in environments where `gettempdir() == "/tmp"` exactly.

## Repro

```bash
TMPDIR=/tmp/sub pytest tests/test_cmd_file.py::test_file_inline_body_staging_round_trip_mismatch_errors
# fails: rc 0
unset TMPDIR; pytest tests/test_cmd_file.py::test_file_inline_body_staging_round_trip_mismatch_errors
# passes (if /tmp is the system tempdir)
```

## Acceptance criteria

- The test exercises the round-trip fence regardless of `$TMPDIR`. Scope the `read_text` interception to the actual staged path — e.g. match against `tempfile.gettempdir()` (`str(self).startswith(tempfile.gettempdir())`) or the specific suffix+location the command uses — rather than a hardcoded `Path("/tmp")`.
- Test passes with `TMPDIR` unset, set to `/tmp`, and set to a `/tmp` subdirectory.
- No production code change required (the fence is correct); if investigation finds otherwise, note it here.

## Out of scope

- The fence logic itself (`jared:610-630`) — verified intact this session.

## Context

Surfaced during #223 (sweep fixture-hygiene). Confirmed pre-existing on `main` by stashing the #223 diff and re-running — unrelated to that change. Filed so the red test on `main` is tracked rather than carried as tribal knowledge.

--- a/tests/test_cmd_file.py
+++ b/tests/test_cmd_file.py
@@ -1,5 +1,6 @@
 import json as _json
 import subprocess as _subprocess
+import tempfile
 from io import StringIO
 from pathlib import Path
 from textwrap import dedent
@@ -301,10 +302,15 @@ def test_file_inline_body_staging_round_trip_mismatch_errors(
     real_read_text = Path.read_text
 
     def maybe_truncating_read_text(self: Path, *a: object, **kw: object) -> str:
-        # Only intercept the temp body file (created in /tmp directly,
-        # not under any subdirectory). The board file lives under
-        # tmp_path/docs/, so we won't break Board parsing.
-        if self.parent == Path("/tmp") and self.suffix == ".md":
+        # Only intercept the staged body file. NamedTemporaryFile creates it
+        # directly in tempfile.gettempdir() (which honors $TMPDIR), so match
+        # the parent against the live tempdir rather than a hardcoded /tmp —
+        # otherwise this fires only when the system tempdir is exactly /tmp
+        # and silently no-ops under TMPDIR=/tmp/sub (e.g. the CC sandbox).
+        # The board file lives under tmp_path/docs/, nested below the tempdir
+        # rather than directly in it, so it stays unintercepted and Board
+        # parsing keeps working.
+        if self.parent == Path(tempfile.gettempdir()) and self.suffix == ".md":
             return ""
         return real_read_text(self, *a, **kw)  # type: ignore[arg-type]
 

